### PR TITLE
Refactor `i18n` to `locale`

### PR DESCRIPTION
**Pull request by Gusted**
- Currently we're using the `i18n` variable naming for the `locale` struct. This contains locale's specific information and cannot be used for general i18n purpose, therefore refactoring it to `locale` makes more sense.
- Ref: https://github.com/go-gitea/gitea/pull/20096#discussion_r906699200

## :warning: BREAKING :warning:

Users who have custom templates needs to replace `.i18n` occurrences with `.locale`.

### CODE REVIEW

File: routers/install/install.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func Init(next http.Handler) http.Handler {
 			Render:  rnd,
 			Session: session.GetSession(req),
 			Data: map[string]interface{}{
-				"i18n":          locale,
+				"Locale":        locale,
```
Comment (wxiaoguang) [inline routers/install/install.go]:
```suggestion
				"locale":        locale,
```

**Review by wxiaoguang — APPROVED** [review]

**Comment by delvh** [comment]
If we are already making breaking changes, we can make it even better:
Is there a usage of `.i18n` in the templates that is not followed by `.Tr*`?
Because if that's the case, we could even take it one step further by simply removing the `.i18n` / `.locale` part.
That has the same breaking effect, is probably easier to use and convert, and requires less unnecessary code...

**Comment by delvh** [comment]
Yeah, then it doesn't exactly make sense to follow my comment above.
But still, should we perhaps offer migrating commands for Unix (something like `sed -ieE 's/\{\{([[:space:]]*)\.i18n\.Tr/{{\1.locale.Tr/g' $CUSTOM_TEMPLATES` (which surprisingly does not compile?)) and Windows (no idea what to do)?

**Comment by jolheiser** [comment]
We could do it as a convenience, but not required imo 
We already don't guarantee compat for templates, specifically because we change them very often.

I wouldn't block this PR for it, personally

**Review by delvh — APPROVED** [review]

**Comment by jolheiser** [comment]
🎺 🤖

**Comment by silverwind** [comment]
This rename causes quite a headache while backporting 😩

**Pull request merged**